src/services/slack.ts

```diff
@@ -20,16 +21,44 @@ export async function sendSlackMessageForEMail(toolkit: Toolkit, emails: string[
             continue;
         }
 
-        const userId = await slackClient.getUserByEmail(email);
+        const user = await slackClient.getUserByEmail(email);
 
-        if (!userId) {
+        if (!user || !user.id) {
             continue;
         }
 
-        await slackClient.sendIMToUser(userId, message);
+        await slackClient.sendIMToUser(user.id, message);
 
         return;
     }
```

Comment:
This loop looks like it should have been replaced with the new method `getSlackUserByEmail` you've implemented.

e.g.:
```typescript
const user = getSlackUserByEmail(toolkit, emails);

if (user !== null) { // or something like "user instanceof User"
  await slackClient.sendIMToUser(user.id, message);
}
```